code/game/objects/items/devices/taperecorder.dm

```diff
@@ -317,19 +323,28 @@
 	storedinfo.Cut()
 	timestamp.Cut()
 
-/obj/item/tape/proc/ruin()
-	ruined = TRUE
-	update_icon(UPDATE_OVERLAYS)
+/obj/item/tape/proc/ruin(mob/user, forced = FALSE)
+	if(forced) // need to be forceable in case it catches on fire to ruin the tape
+		ruined = TRUE
+		update_icon(UPDATE_OVERLAYS)
+	else
+		to_chat(user, "<span class='notice'>You start pulling the tape out.</span>")
+		if(do_after(user, 1 SECONDS, target = src))
+			to_chat(user, "<span class='notice'>You pull the tape out of [src].</span>")
+			ruined = TRUE
+			update_icon(UPDATE_OVERLAYS)
```

Comment:
you only ever set forced to `TRUE` if you dont pass a user arg. couldn't you just do if(!user) instead of if(forced)